omnigibson/systems/macro_particle_system.py

```diff
@@ -43,14 +29,13 @@ class MacroParticleSystem(BaseSystem):
     # Template object to use -- this should be some instance of BasePrim. This will be the
     # object that symbolizes a single particle, and will be duplicated to generate the particle system.
     # Note that this object is NOT part of the actual particle system itself!
-    particle_object = None
+    _particle_object = None
 
     # dict, array of particle objects, mapped by their prim names
     particles = None
 
-    # Scaling factor to sample from when generating a new particle
-    min_scale = None              # (x,y,z) scaling
-    max_scale = None              # (x,y,z) scaling
+    # Counter to increment monotonically as we add more particles
+    _particle_counter = None
```

Comment:
If we want to manually keep track of `_particle_counter`, than we also need to add it to load/dump_state?

Reply:
done